### PR TITLE
Add create addon cmd

### PR DESCRIPTION
**Pull request by qiujian16**

### CODE REVIEW

**Comment by qiujian16** [comment]
/assign @zhujian7 
/hold 
still in progress

File: pkg/cmd/addon/create/cmd.go [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+			}
+			if err := o.Run(); err != nil {
+				return err
+			}
+
+			return nil
+		},
+	}
+
+	cmd.Flags().StringVarP(&o.Version, "name", "v", "0.0.1", "Specified version of the addon")
```
Comment (zhujian7) [inline pkg/cmd/addon/create/cmd.go:49]:
```
clusteradm addon create --help

panic: unable to redefine 'v' shorthand in "create" flagset: it's already used for "name" flag
```

The `-v` here conflicts with the global option `-v`(number for the log level verbosity)

Comment (zhujian7) [reply pkg/cmd/addon/create/cmd.go:49]:
```suggestion
	cmd.Flags().StringVar(&o.Version, "version", "0.0.1", "Specified version of the addon")
```

File: pkg/cmd/addon/create/exec.go [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+		},
+	}
+
+	return addon, nil
+}
+
+func newClusterManagementAddon(o *Options) *addonv1alpha1.ClusterManagementAddOn {
+	cma := &addonv1alpha1.ClusterManagementAddOn{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: o.Name,
```
Comment (zhujian7) [inline pkg/cmd/addon/create/exec.go:44]: Should we add the annotation:
```
addon.open-cluster-management.io/lifecycle: "addon-manager"
```

File: pkg/cmd/addon/create/options.go [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+	// version is the version of the addon
+	Version string
+
+	Overwrite bool
+
+	FileNameFlags genericclioptions.FileNameFlags
+	//
+	Streams genericclioptions.IOStreams
+
+	Placements []string
```
[the remark this answers is not in the document]
Comment (qiujian16) [reply pkg/cmd/addon/create/options.go]: Rethinking it a bit, I do not think we should set installstartegy on create command, maybe a config cmd later would be better.

**Comment by zhujian7** [comment]
@qiujian16  could you also add some e2e for this?

**Pull request merged**